### PR TITLE
update: infinite potion duration support

### PR DESCRIPTION
**Pull request by DeidaraMC**
Added an INFINITE_DURATION constant in Potion and updated Entity#effectTick to support it

### CODE REVIEW

**Comment by MelonHell** [comment]
resolves #91

File: src/main/java/net/minestom/server/potion/Potion.java [hunk 1]
```diff
@@ -48,6 +48,11 @@ public record Potion(@NotNull PotionEffect effect, byte amplifier,
      */
     public static final byte ICON_FLAG = 0x04;
 
+    /**
+     * a duration constant which sets aqgit s Potion duration to infinite.
```
Comment (Codestech1) [inline src/main/java/net/minestom/server/potion/Potion.java:52]: I think the `a` letter at a beginning of the javadoc should be upper case.

File: src/main/java/net/minestom/server/entity/Entity.java [hunk 2]
```diff
@@ -724,7 +724,9 @@ private void effectTick(long time) {
         final List<TimedPotion> effects = this.effects;
         if (effects.isEmpty()) return;
         effects.removeIf(timedPotion -> {
-            final long potionTime = (long) timedPotion.getPotion().duration() * MinecraftServer.TICK_MS;
+            long duration = timedPotion.getPotion().duration();
+            if (duration == Potion.INFINITE_DURATION) return false;
+            final long potionTime = duration * MinecraftServer.TICK_MS;
```
Comment (Codestech1) [inline src/main/java/net/minestom/server/entity/Entity.java:729]: Add a space here, just my opinion.

File: src/main/java/net/minestom/server/potion/Potion.java (hunk 1 above)
Comment (Codestech1) [inline src/main/java/net/minestom/server/potion/Potion.java:52]: I think the `a` letter at a beginning of the javadoc should be upper case.

**Review by Codestech1 — APPROVED** [review]
LGTM

File: src/main/java/net/minestom/server/potion/Potion.java (hunk 1 above)
Comment (mworzala) [inline src/main/java/net/minestom/server/potion/Potion.java:52]: I agree on capitalizing the A, also `aqgit s`

**Review by mworzala — CHANGES_REQUESTED** [review]

File: src/main/java/net/minestom/server/entity/Entity.java (hunk 2 above)
Comment (mworzala) [reply src/main/java/net/minestom/server/entity/Entity.java:729]: No, there should not be an extra space on this line

**Pull request merged**